src/ioctls/vm.rs

```diff
@@ -394,9 +393,22 @@ impl VmFd {
     /// # use kvm_bindings::{kvm_userspace_memory_region, KVM_MEM_LOG_DIRTY_PAGES};
     /// # let kvm = Kvm::new().unwrap();
     /// # let vm = kvm.create_vm().unwrap();
-    /// // This examples is based on https://lwn.net/Articles/658511/.
-    /// let mem_size = 0x4000;
-    /// let guest_addr: u64 = 0x1000;
+    /// let mut mem_size = 0x0;
+    /// let mut guest_addr = 0x0;
+    ///
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// {
+    ///     // This examples is based on https://lwn.net/Articles/658511/.
```

Comment:
This comment feels a bit out of place here before the initialization of guest memory. Can we move it at the beginning of the example and say something like for x86_64 the example is based on bla bla?